api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -142,6 +142,13 @@ object ServiceReader {
             }
           case _ => throw new IllegalStateException("ServiceCall does not have 3 type arguments?")
         }
+      } else if (ScalaSig.isScala(serviceInterface)) {
+        if (serviceInterface.isInterface()) {
+          val implClass = Class.forName(serviceInterface.getName + "$class", false, classLoader)
+          val method = implClass.getMethod(DescriptorMethodName, serviceInterface)
+          method.invoke(null, proxy.asInstanceOf[AnyRef])
+        } else
+          throw new IllegalArgumentException("Service.descriptor must be implemented in a trait")
```

Comment:
Ok, I worked it out, the method is typically the one declared on the service interface (or proxy even), not on `Service`, so they don't match.